### PR TITLE
enable upload with envvar or Spark property

### PR DESCRIPTION
**Pull request by cseed**

### CODE REVIEW

File: hail/src/main/scala/is/hail/HailContext.scala [hunk 1]
```diff
@@ -247,6 +247,12 @@ object HailContext {
     val hc = new HailContext(sparkContext, sqlContext, logFile, hailTempDir, branchingFactor)
     sparkContext.uiWebUrl.foreach(ui => info(s"SparkUI: $ui"))
 
+    var pipelineEmail = System.getenv("HAIL_PIPELINE_UPLOAD_EMAIL")
+    if (pipelineEmail == null)
+      pipelineEmail = sparkContext.getConf.get("hail.pipelineUploadEmail", null)
+    if (pipelineEmail != null)
+      hc.enablePipelineUpload(pipelineEmail)
```
Comment (tpoterba) [inline hail/src/main/scala/is/hail/HailContext.scala]: we should definitely warn here.

Comment (cseed) [reply hail/src/main/scala/is/hail/HailContext.scala]: I warn (used to info) in enablePipelineUpload.  Is that insufficient?

**Comment by cseed** [comment]
I separated setting the email from enabling pipeline uploads, so people can set the email once and use hl.upload_log() without arguments independent of pipeline upload.

File: hail/src/main/scala/is/hail/HailContext.scala (hunk 1 above)
Comment (cseed) [reply hail/src/main/scala/is/hail/HailContext.scala]: Also, note, for some reason those warns are going into the log file but not the console, in spite of the fact they are running after configureLogging.  Still looking into it.

File: hail/python/hail/context.py [hunk 2]
```diff
@@ -277,21 +277,36 @@ def read_version_info() -> str:
...
+    If email is not set, uploads will be anonymous.  Upload email can
+    also be set through the `HAIL_UPLOAD_EMAIL` environment variable
+    or the `hail.uploadEmail` Spark configuration property.
 
     Parameters
     ----------
     email : :obj:`str`
         Email contact to include with uploaded data.
+
     """
```
[the remark this answers is not in the document]
Comment (cseed) [reply hail/python/hail/context.py]: Oops!  Fixed.  I also added a (private) `_set_upload_url` for (eventual) testing of the upload service.

**Pull request merged**